File: src/encoders/pronote-date.ts
/**
 * @returns A string formatted as `$d/$m/$yyyy $H:$M:$S`.
 * @example
 * encodePronoteDate(new Date());
 * // -> "21/1/2024 0:0:0"
 */
export const encodePronoteDate = (date: Date): string => {
  const day = date.getDate();
  const month = (date.getMonth() + 1);
  const year = date.getFullYear();

  const hours = date.getHours();
  const minutes = date.getMinutes();
  const seconds = date.getSeconds();

  return `${day}/${month}/${year} ${hours}:${minutes}:${seconds}`;
};

File: library/src/commonMain/kotlin/encoders/pronote-date.kt
package encoders

import kotlinx.datetime.LocalDateTime

/**
 * @returns A string formatted as `$d/$m/$yyyy $H:$M:$S`.
 */

fun encodePronoteDate (date: LocalDateTime): String {
    val day = date.dayOfMonth
    val month = date.month.value
    val year = date.year

    val hours = date.hour
    val minutes = date.minute
    val seconds = date.second

    return "$day/$month/$year $hours:$minutes:$seconds"
}